src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -620,7 +604,7 @@ internal static string GetCookieHeader(Uri uri, CookieContainer cookies)
             // Manually add cookies.
             if (cookies != null)
             {
-                string cookieHeader = GetCookieHeader(requestMessage.RequestUri, cookies);
+                string cookieHeader = WinHttpCookieHelper.GetCookieHeader(requestMessage.RequestUri, cookies);
```

Comment:
The calculation of the 'Cookie' header value is already in the adapter class.  Moving the logic of appending the request header itself seems like a stretch.  Then the adapter has to know about appending headers in general which seems out of scope.